# Project Title: 
Telecommunications Customer Churn Prediction Analysis
Covid-19 Vaccination Sentiment Analysis With National Language Processing

# Business Understanding

# Setup

## Importation

In [1]:
# Import necessary libraries
import numpy as np 
import pandas as pd 

## Loading The Datasets

In [2]:
# Load the train dataset
train_df = pd.read_csv('Train.csv')
train_df.head()

tweet_id                                          safe_text  label  \
0  CL1KWCMY  Me &amp; The Big Homie meanboy3000 #MEANBOY #M...    0.0   
1  E3303EME  I'm 100% thinking of devoting my career to pro...    1.0   
2  M4IVFSMS  #whatcausesautism VACCINES, DO NOT VACCINATE Y...   -1.0   
3  1DR6ROZ4  I mean if they immunize my kid with something ...   -1.0   
4  J77ENIIE  Thanks to <user> Catch me performing at La Nui...    0.0   

   agreement  
0        1.0  
1        1.0  
2        1.0  
3        1.0  
4        1.0

In [3]:
# Load the train dataset
test_df = pd.read_csv('Test.csv')
test_df.head()

tweet_id                                          safe_text
0  00BHHHP1  <user> <user> ... &amp; 4 a vaccine given 2 he...
1  00UNMD0E  Students starting school without whooping coug...
2  01AXPTJF  I'm kinda over every ep of <user> being "rippe...
3  01HOEQJW  How many innocent children die for lack of vac...
4  01JUKMAO  CDC eyeing bird flu vaccine for humans, though...

# Exploratory Data Analysis: EDA

### Column Information of The Datasets

In [4]:
train_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10001 entries, 0 to 10000
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   tweet_id   10001 non-null  object 
 1   safe_text  10001 non-null  object 
 2   label      10000 non-null  float64
 3   agreement  9999 non-null   float64
dtypes: float64(2), object(2)
memory usage: 312.7+ KB


- The train DataFrame contains a total of 10,001 entries (rows), and four columns: 'tweet_id', 'safe_text', 'label', and 'agreement'.
- The 'tweet_id' and 'safe_text' columns are of type 'object', typically representing strings. The 'label' and 'agreement' columns are of type 'float64', indicating floating-point numbers.
- The 'label' and 'agreement' columns contain missing values.

In [5]:
test_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5177 entries, 0 to 5176
Data columns (total 2 columns):
 #   Column     Non-Null Count  Dtype 
---  ------     --------------  ----- 
 0   tweet_id   5177 non-null   object
 1   safe_text  5176 non-null   object
dtypes: object(2)
memory usage: 81.0+ KB


- The test DataFrame consists of 5,177 entries (rows), and two columns: 'tweet_id' and 'safe_text'.
- Both columns, 'tweet_id' and 'safe_text', are of type 'object', representing strings.
- The 'safe_text' column has a missing value.

### Column Information of The Datasets

In [18]:
# Checking the shape of the train dataset
train_df.shape

(10001, 4)

In [19]:
# Checking the shape of the train dataset
test_df.shape

(5177, 2)

In [21]:
# Checking for missing values in the train dataset
train_df.isnull().sum()

tweet_id     0
safe_text    0
label        1
agreement    2
dtype: int64

In [23]:
# Handle missing values in the train dataset
train_df['label'].fillna(0, inplace=True)  # Fill missing label with 0 (neutral sentiment)
train_df['agreement'].fillna(train_df['agreement'].mean(), inplace=True)  # Fill missing agreement with mean

print("Missing values in train dataset:")
print(train_df.isnull().sum())

Missing values in train dataset:
tweet_id     0
safe_text    0
label        0
agreement    0
dtype: int64


In [24]:
# Checking for missing values in the train dataset
test_df.isnull().sum()

tweet_id     0
safe_text    0
dtype: int64

In [25]:
# Handle missing values in the test dataset
test_df['safe_text'].fillna('', inplace=True)

# Confirm no more missing values
print("Missing values in test dataset:")
print(test_df.isnull().sum())

Missing values in test dataset:
tweet_id     0
safe_text    0
dtype: int64
